pandas/tests/io/excel/test_readers.py

```diff
@@ -148,6 +149,26 @@ def parser(self, *args, **kwargs):
             expected = expected_defaults[read_ext[1:]]
         assert result == expected
 
+    def test_engine_kwargs(self, read_ext, engine):
+        # GH#52214
+        expected_defaults = {
+            "xlsx": {"foo": "abcd"},
+            "xlsm": {"foo": 123},
+            "xlsb": {"foo": "True"},
+            "xls": {"foo": True},
+            "ods": {},
+        }
+
+        msg = re.escape(r"read_excel() got an unexpected keyword argument 'foo'")
```

Comment:
This is not the expected error message - it should be when pandas calls the engine with an incorrect kwarg that should raise. Below, the line should be `engine_kwargs=expected_defaults[...]`.